service/file.go

```diff
@@ -46,3 +48,34 @@ func FileCopy(src *model.Matter, parent string) error {
 	nm.Parent = parent
 	return gormutil.DB().Create(nm).Error
 }
+
+func CanMove(uid int64, parent string, file *model.Matter) (bool, error) {
+	var (
+		out []model.Matter
+	)
+	//check dir exists
+	if !MatterParentExist(uid, parent) {
+		return false, errors.New("dir does not exists")
+	}
+	//avoid move to itself
+	//eg: a/ -> a/
+	//eg: a/ -> a/b/c
+	//eg: b/ -> a/b/c
+	if parent != "" && strings.HasPrefix(parent, file.Parent+file.Name) {
```

Comment:
`parent + file.name == file.parent`

Just judge whether the target directory is equal to the original directory, right?

Reply:
uh no,this condition not only avoid folder move to itself,but also avoid folder move to it's children folder.

`//eg: a/ -> a/`  means: move folder `a` to itself . so  `parent + file.name == file.parent`  =>  `a/a == ''`